e2e/nomostest/clusters/gke.go

```diff
@@ -21,26 +21,34 @@ import (
 	"strings"
 	"time"
 
+	"github.com/pkg/errors"
 	"kpt.dev/configsync/e2e"
-	"kpt.dev/configsync/e2e/nomostest/ntopts"
 	"kpt.dev/configsync/e2e/nomostest/testing"
 )
 
-// GKECluster tells the test to use the GKE cluster pointed to by the config flags.
-func GKECluster(t testing.NTB) ntopts.Opt {
-	return func(opt *ntopts.New) {
-		t.Helper()
-		clusterName := *e2e.GCPCluster
-		if *e2e.CreateClusters {
-			opt.IsEphemeralCluster = true
-			clusterName = opt.Name
-			t.Cleanup(func() {
-				deleteGKECluster(t, clusterName)
-			})
-			createGKECluster(t, clusterName)
-		}
-		getGKECredentials(t, clusterName, opt.KubeconfigPath)
-	}
+// GKECluster is a GKE cluster for use in the e2e tests
+type GKECluster struct {
+	// T is a testing interface
+	T testing.NTB
+	// Name is the name of the cluster
+	Name string
+	// KubeConfigPath is the path to save the kube config
+	KubeConfigPath string
+}
+
+// Create the GKE cluster
+func (c *GKECluster) Create() error {
+	return createGKECluster(c.T, c.Name)
```

Comment:
can these functions just be inlined?

Reply:
They could - I opted to call out to the nested functions because I figured it might help with readability. But I can move the implementation inline